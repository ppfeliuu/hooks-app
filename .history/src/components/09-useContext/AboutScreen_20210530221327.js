import React from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setuser } = useContext(UserContext);
  return (
    <div>
      <h1>About Screen</h1>
      <hr />
    </div>
  );
};
