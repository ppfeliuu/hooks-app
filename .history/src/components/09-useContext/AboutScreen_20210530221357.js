import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setuser } = useContext(UserContext);
  return (
    <div>
      <h1>About Screen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};
