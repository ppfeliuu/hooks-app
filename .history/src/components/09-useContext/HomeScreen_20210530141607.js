import React from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const userContext = UserContext(contextValue);
  console.log(contextValue);
  return (
    <div>
      <h1>Home Screen</h1>
      <hr />
    </div>
  );
};
