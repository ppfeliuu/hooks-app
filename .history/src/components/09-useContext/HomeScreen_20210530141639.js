import React, { useContext } from "react";

export const HomeScreen = () => {
  const userContext = useContext(contextValue);
  console.log(contextValue);
  return (
    <div>
      <h1>Home Screen</h1>
      <hr />
    </div>
  );
};
