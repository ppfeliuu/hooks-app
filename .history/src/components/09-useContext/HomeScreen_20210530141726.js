import React, { useContext } from "react";

export const HomeScreen = () => {
  const userContext = useContext(contextValue);
  console.log(userContext);

  return (
    <div>
      <h1>Home Screen</h1>
      <hr />
    </div>
  );
};
