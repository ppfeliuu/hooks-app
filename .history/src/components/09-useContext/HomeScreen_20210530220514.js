import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  console.log();

  return (
    <div>
      <h1>Home Screen</h1>
      <hr />
    </div>
  );
};
