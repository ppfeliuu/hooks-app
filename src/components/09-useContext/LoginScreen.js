import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setuser } = useContext(UserContext);
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />

      <button
        className="btn btn-primary"
        onClick={() =>
          setuser({
            id: 1,
            name: "Pedro",
          })
        }
      >
        Login
      </button>
    </div>
  );
};
