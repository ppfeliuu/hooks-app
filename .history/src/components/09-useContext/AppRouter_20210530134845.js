import React from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route path="/" exact component={HomeScreen} />
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
        </Switch>
      </div>
    </Router>
  );
};
