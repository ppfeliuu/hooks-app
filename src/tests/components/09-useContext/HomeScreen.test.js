import { shallow, mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("Test home screen", () => {
  const user = {
    name: "Pedro",
    email: "pedro@gmail.com",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );
  test("should show component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
