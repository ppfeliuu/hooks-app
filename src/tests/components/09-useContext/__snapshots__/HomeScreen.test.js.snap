// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home screen should show component 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Pedro",
   "email": "pedro@gmail.com"
}
  </pre>
</div>
`;
